interface.go

```diff
@@ -279,7 +279,7 @@ func (f *Interface) listenOut(i int) {
 	nb := make([]byte, 12, 12)
 
 	li.ListenOut(func(fromUdpAddr netip.AddrPort, payload []byte) {
-		f.readOutsidePackets(fromUdpAddr, nil, plaintext[:0], payload, h, fwPacket, lhh, nb, i, ctCache.Get(f.l))
```

Comment:
does this add an allocation in the hot path?

Reply:
It doesn't escape, at least currently. I asked Jack to run a perf test on both `struct` vs` *struct` commits I have here. I expect `*struct` is maybe a tiny fraction faster, we would just need to be careful not to have it escape in future commits.